### PR TITLE
Remove sass lint

### PR DESCRIPTION
**Pull request by vanitabarrett**
Dependant on https://github.com/alphagov/govuk-frontend/pull/1875
Needs to be rebased against master once that's merged

Commits to review:
 
- [Remove sass lint](https://github.com/alphagov/govuk-frontend/pull/1902/commits/529f2b9285233c6c06cbef70f3a19dc2a5a4a3ed)
- [Remove outdated docs on gulp:lint task](https://github.com/alphagov/govuk-frontend/pull/1902/commits/098a8db1aa72b0971aeb3722359eb9e51a9fd4dd)
- [Remove old sass-lint disable rules](https://github.com/alphagov/govuk-frontend/pull/1902/commits/46ca779aa203aeecee38147e463eb14826b9c755)
- [Update docs](https://github.com/alphagov/govuk-frontend/pull/1902/commits/a562675f3e04a810aa85fb796d1a430d7dfed213)
- [Tweak font-faces file to appease linter](https://github.com/alphagov/govuk-frontend/pull/1902/commits/a3c236ca109e3619e9d1434a1197c999ab1b24d1)

Closes https://github.com/alphagov/govuk-frontend/issues/1898

### CODE REVIEW

File: docs/contributing/coding-standards/css.md [hunk 1]
```diff
@@ -112,19 +112,13 @@ This makes it easier to keep track of different contexts.
 # Linting
 
 To ensure code quality and consistency in our Sass files we check that certain
-style rules are followed using a project [YAML file](../../../config/.sass-lint.yml)
-
-As we're using node-sass parser to parse our scss files, the package of choice
-is [sass-lint](https://github.com/sasstools/sass-lint).
+style rules are followed. These rules are based on [stylelint-config-gds](https://github.com/alphagov/stylelint-config-gds/blob/main/scss.js), but we also add our own custom rules with a project [YAML file](../../../.stylelintrc.yml)
```
Comment (m-green) [inline docs/contributing/coding-standards/css.md:115]:
```suggestion
style rules are followed. These rules are based on [stylelint-config-gds](https://github.com/alphagov/stylelint-config-gds/blob/main/scss.js), but we also add our own custom rules with a project [YAML file](../../../.stylelintrc.yml).
```

File: docs/contributing/coding-standards/css.md [hunk 2]
```diff
@@ -112,19 +112,13 @@ This makes it easier to keep track of different contexts.
...
 
 See [testing and linting](/docs/releasing/testing-and-linting.md) for more information.
 
 ## Running the lint task
 
-You can run the linter in [gulp](https://www.npmjs.com/package/gulp-sass-lint)
-or check linting directly in [Sublime Text](https://github.com/skovhus/SublimeLinter-contrib-sass-lint),
-[Atom](https://atom.io/packages/linter-sass-lint)
-or other [IDE's](https://github.com/sasstools/sass-lint#ide-integration)
+You can run the linter in gulp: `gulp scss:stylelint` or check linting directly in [Sublime Text](https://github.com/SublimeLinter/SublimeLinter-stylelint), [Atom](https://atom.io/packages/linter-stylelint) or other [IDE's](https://stylelint.io/user-guide/integrations/editor)
```
Comment (m-green) [inline docs/contributing/coding-standards/css.md:121]:
```suggestion
You can run the linter in gulp by running `gulp scss:stylelint`, or use linting in [Sublime Text](https://github.com/SublimeLinter/SublimeLinter-stylelint), [Atom](https://atom.io/packages/linter-stylelint) or [other editors that support stylelint](https://stylelint.io/user-guide/integrations/editor).
```

File: docs/contributing/coding-standards/css.md [hunk 3]
```diff
@@ -441,7 +435,7 @@ Good:
 $my-example-var: value;
 ```
 
-### Don't write leading or trailing zeros for numeric values with a decimal point
+### Don't write trailing zeros for numeric values with a decimal point
```
Comment (m-green) [inline docs/contributing/coding-standards/css.md]:
```suggestion
### Don't write trailing zeroes for numeric values with a decimal point
```

**Review by m-green — COMMENTED** [review]
Looks good to me! Just a couple of style suggestions and fixes.

File: docs/contributing/tasks.md [hunk 4]
```diff
@@ -54,7 +54,7 @@ This task will:
 **`gulp test`**
 
 This task will:
-- Run scss:lint
+- Run scss:stylelint
```
Comment (36degrees) [inline docs/contributing/tasks.md:57]: Given we no longer need to run both tasks side-by-side, can we rename the stylelint task to just 'lint', effectively preserving the existing take name and just updating it to use stylelint?

File: docs/contributing/coding-standards/css.md [hunk 5]
```diff
@@ -453,13 +447,13 @@ Bad:
...
 .selector {
-  font-size: 0.5em;
+  font-size: .5em;
 }
 ```
 
 ### Remove trailing whitespace
 
-More write up on [supported rules](https://github.com/sasstools/sass-lint/tree/master/docs/rules).
+More write up on [supported rules](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configure.md#rules).
```
Comment (36degrees) [inline docs/contributing/coding-standards/css.md:456]: I'm not entirely sure what the need for this link is, but wondering if https://stylelint.io/user-guide/rules/about or https://stylelint.io/user-guide/rules/list might be a better place to link to – the current URL seems to be more about configuring rules or adding entirely new ones.

File: src/govuk/helpers/_font-faces.scss [hunk 6]
```diff
@@ -19,25 +17,25 @@
   @include govuk-not-ie8 { // In IE8, which cannot render WOFF format, we fall back to system fonts
     @include govuk-exports("govuk/helpers/font-faces") {
       @at-root {
-        /*! Copyright (c) 2011 by Margaret Calvert & Henrik Kubel. All rights reserved. The font has been customised for exclusive use on gov.uk. This cut is not commercially available. */
+        //! Copyright (c) 2011 by Margaret Calvert & Henrik Kubel. All rights reserved. The font has been customised for exclusive use on gov.uk. This cut is not commercially available.
```
Comment (36degrees) [inline src/govuk/helpers/_font-faces.scss:20]: Just to double check – this copyright notice still appears in the compiled CSS?

Comment (vanitabarrett) [reply src/govuk/helpers/_font-faces.scss:20]: Yep 👍 I ran `npm run build:package` and the file still contains the notice:

Screenshot 2020-08-25 at 11 57 34

Comment (36degrees) [reply src/govuk/helpers/_font-faces.scss:20]: And in the compiled CSS in `dist/` as well?

Comment (vanitabarrett) [reply src/govuk/helpers/_font-faces.scss]: Good catch, it wasn't being included there. Looks like you can only include comments with `/*!` so I've reverted that line and included `stylelint-disable scss/comment-no-loud`

Comment (36degrees) [reply src/govuk/helpers/_font-faces.scss]: :+1:

I'm wondering if that's worth writing a test for in `after-build-dist.test.js` – I think it's pretty important that the copyright notice stays intact. Do you think it'd make sense to do that as part of this PR?

**Comment by vanitabarrett** [comment]
@36degrees Added a test for the copyright notice 🙂

**Review by 36degrees — APPROVED** [review]
I've only reviewed 529f2b9285233c6c06cbef70f3a19dc2a5a4a3ed through d0392d86ef2e2052e6c97ce302f14b26c77f2ea3, as the other commits have been reviewed in #1875.

**Pull request merged**